plugins/chromatic_correction.py

```diff
@@ -0,0 +1,355 @@
+                "excitation": [float(input_wl[0]), float(input_wl[1])],
+                "emission": [float(output_wl[0]), float(output_wl[1])],
+                "exposure time": float(da.metadata[model.MD_EXP_TIME]),
+                "light power": float(da.metadata[model.MD_LIGHT_POWER])
+            }
+
+            # Save the modified image and display
+            file_location = get_file_location(f"{da.metadata[model.MD_DESCRIPTION]}_modified_metadata",
+                                              "ome.tiff")
+            self.display_static_stream(file_location, da)
+
+        return transformation_dict
+
+    def open_dialog_window(self):
+        """Opens a dialog window which shows start button if all the specific requirements are present
+        otherwise suggests changing the settings"""
+        localization_tab = self.main_app.main_data.getTabByName("cryosecom-localization")
+        tab_data = localization_tab.tab_data_model
+        all_emission = [numpy.array(fms.emission.value) for fms in
+                        tab_data.streams.value]
+        # reference channel should be present in the present streams
+        check_ref_channel = any(
+            all(numpy.isclose(emission, self.ref_emission, rtol=1e-09, atol=1e-09)) for emission in
+            all_emission)
+        # No two streams should have same emission
+        unique_elements = set(map(tuple, all_emission))
+        check_unique_channels = len(unique_elements) == len(all_emission)
+        self.dlg = AcquisitionDialog(self, "Chromatic Correction",
+                                     ("Ready to start Chromatic aberration on the given channel.\n"
+                                      "The process will take about 5 minutes\n\n"
+                                      "Z stack range is {} m\n"
+                                      "Z step size of {} m".format(Z_RANGE, Z_STEP_SIZE)))
+        if (len(tab_data.streams.value) <= 1) or (not check_ref_channel) or (not check_unique_channels):
+            txt = ("Problems\n1. Add minimum of two streams.\n2. One of the streams must be green channel.\n"
+                   "3. No two streams should be the same.")
+            self.dlg.lbl_description.SetLabel(txt)
+            self.dlg.addButton("Manually fix")
+            self.dlg.ShowModal()
+            return
+        self.dlg.Bind(wx.EVT_CLOSE, self.on_close)
+        self.dlg.addButton("Start", self._on_start, face_colour='blue')
+        self.dlg.ShowModal()
+
+    def on_close(self, evt, *args, **kwargs):
+        self.dlg.buttons[0].SetLabel("Cancelling...")
+        self.dlg.buttons[0].Disable()
+        self.dlg.lbl_description.SetLabel("Cancellation in progress\nMay take some time to "
+                                          "cancel the already running processes")
+        self.dlg.Destroy()
+        if self.acqui_task:
+            self.acqui_task.cancel()
+        logging.debug("Dialog closed")
+
+    def _on_start(self, evt):
```

Comment:
For the plugin buttons, the callback takes the AcquisitionDialog as argument (so `evt` -> `dlg`), and runs *in a separate thread*.
So you shouldn't do anything related with the GUI directly in this function, or protect it with `call_in_wx_main()`. That second option is probably what you want.

Reply:
I looked into timelapse.py and modified the code. It takes dlg as input argument. Changed self.dlg to dlg, without the use of the decorator. Would that be a fine idea?

Reply:
When I protect _on_start with call_in_wx_main(), it seems to block the functioning by popping up Odemis is not responding message box

Reply:
I have added the decorator on on_close(), as it was losing reference of the dialog box